prow/plugins/plugins.go

```diff
@@ -190,6 +190,31 @@ func (a *Agent) CommentPruner() (*commentpruner.EventClient, error) {
 	return a.commentPruner, nil
 }
 
+func (a *Agent) Presubmits(pr *github.PullRequest) (string, []config.Presubmit, error) {
+	baseSHA, presubmits, err := a.presubmits(pr)
+	if err != nil {
+		// log error
+		// Render error comment and create it on Github if it doesn't exist already
+		return "", nil, err
+	}
+	// Check if error comment exists on Github and if yes, remove it
+	return baseSHA, presubmits, nil
+}
+
+func (a *Agent) presubmits(pr *github.PullRequest) (string, []config.Presubmit, error) {
+	org, repo := pr.Base.Repo.Owner.Login, pr.Base.Repo.Name
+	baseSHA, err := c.GitHubClient.GetRef(org, repo, "heads/"+pr.Base.Ref)
```

Comment:
I don't think we should be looking up the base SHA here. I think in most or all cases the caller will already have a specific base SHA value to use (this is at least true for the `trigger` and `tide` cases) so lets just pass it in.

edit: I see now that you are having this function be responsible for determining the baseSHA to use and then reusing that value. I don't think this really adds any value over `config.GetPresubmits()` except for the GH comment generation and pruning, but I also don't think we want a generic error comment for all uses of this. 
I don't feel super strongly about this, but this seems out of place in the plugin agent and I'm not sure it adds anything besides overly generalized error commenting.

Reply:
> all plugins need to look up the baseSHA and have logic for error reporting if something fails.

Only plugins that actually need the in repo config functionality. Either way I don't think thats worth avoiding, its only a few line of common go boilerplate and I think the flow makes more sense that way.
Moving this to the `config` package (or just using `config.GetPresubmits()`) would make a lot more sense IMO. Things like `tide` shouldn't have to import the `plugins` package to use this logic. Having to pass some extra args is totally fine.

Reply:
> Only plugins that actually need the in repo config functionality.

Everything that accesses the `Presubmits` or not? I thought we settled on integrating this everywhere so we don't get weird issues because different components have a different view on the current state of config.

> Moving this to the config package (or just using config.GetPresubmits()) would make a lot more sense IMO. Things like tide shouldn't have to import the plugins package to use this logic. Having to pass some extra args is totally fine.

Well, tide works just fine with what is in the `config` package. But I am fine moving it :+1: